modules/tinymce/src/core/test/ts/browser/keyboard/HomeEndKeysTest.ts

```diff
@@ -53,6 +53,25 @@ describe('browser.tinymce.core.keyboard.HomeEndKeysTest', () => {
       TinyAssertions.assertSelection(editor, [ 0, 0 ], 1, [ 0, 0 ], 1);
     });
 
+    context('Inline element', () => {
+      it('move caret at the begining on the line', () => {
+        const editor = hook.editor();
+        editor.setContent('<p>test <a href="google.com">link</a> test</p>');
+        TinySelections.setCursor(editor, [ 0, 1, 0 ], 0);
+        TinySelections.setSelection(editor, [ 0, 1, 0 ], 0, [ 0, 1, 0 ], 5);
+        TinyContentActions.keystroke(editor, Keys.home());
+        TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
+      });
+
+      it('move caret at the beginning on the line (parent) if the first element is an inline element', () => {
```

Comment:
I think you only need to check the cases where there is a link at the start of the element. So hitting home inside plain text after a link should move the caret before the link element. Hitting home inside the middle of a link should move the caret to before the link. Something like that.

Reply:
Done